contracts/marketplace/Marketplace.sol

```diff
@@ -344,6 +344,17 @@ contract Marketplace is
 
         emit ListingRemoved(_listingId, targetListing.tokenOwner);
     }
+    
+     /// @dev Returns all marketplace listings.
+     function getAllListing() external view returns (Listing[] memory) {
+         uint256 listingCount = totalListings;
+         Listing[] memory allListings = new Listing[](listingCount);
+
+         for (uint index=0; index < listingCount; index++){
+             allListings[index] = listings[index];
+         }
+         return allListings;
+     }
```

Comment:
Hi @Aymen1001  wanted to ask why not just do this, is this any extra gas usage in this implementation?
```
  function getAllListing() external view returns (Listing[] memory) {
         Listing[] memory allListings = new Listing[](totalListings);

         for (uint index=0; index < totalListings; index++){
             allListings[index] = listings[index];
         }
         return allListings;
     }
```

Reply:
As it's a view function both don't cost gas, I just got the habit of writing it that way